specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2022-09-01/AppServiceCertificateOrders.json

```diff
@@ -0,0 +1,1801 @@
+            "isPrivateKeyExternal": {
+              "description": "Should we change the ASC type (from managed private key to external private key and vice versa).",
+              "type": "boolean"
+            }
+          },
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "RenewCertificateOrderRequest": {
+      "description": "Class representing certificate renew request.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../Microsoft.Web/stable/2022-09-01/CommonDefinitions.json#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "RenewCertificateOrderRequest resource specific properties",
+          "type": "object",
+          "properties": {
+            "keySize": {
+              "format": "int32",
+              "description": "Certificate Key Size.",
+              "type": "integer"
+            },
+            "csr": {
+              "description": "Csr to be used for re-key operation.",
+              "type": "string"
+            },
+            "isPrivateKeyExternal": {
+              "description": "Should we change the ASC type (from managed private key to external private key and vice versa).",
+              "type": "boolean"
+            }
+          },
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "SiteSeal": {
+      "description": "Site seal",
+      "required": [
+        "html"
+      ],
+      "type": "object",
+      "properties": {
+        "html": {
+          "description": "HTML snippet",
+          "type": "string",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "SiteSealRequest": {
+      "description": "Site seal request.",
+      "type": "object",
+      "properties": {
+        "lightTheme": {
+          "description": "If <code>true</code> use the light color theme for site seal; otherwise, use the default color theme.",
+          "type": "boolean"
+        },
+        "locale": {
+          "description": "Locale of site seal.",
+          "type": "string"
+        }
+      }
+    }
+  },
+  "parameters": {
+    "subscriptionIdParameter": {
+      "name": "subscriptionId",
+      "in": "path",
+      "description": "Your Azure subscription ID. This is a GUID-formatted string (e.g. 00000000-0000-0000-0000-000000000000).",
+      "required": true,
+      "type": "string"
+    },
+    "resourceGroupNameParameter": {
+      "name": "resourceGroupName",
+      "in": "path",
+      "description": "Name of the resource group to which the resource belongs.",
+      "required": true,
+      "type": "string",
+      "maxLength": 90,
+      "minLength": 1,
+      "pattern": "^[-\\w\\._\\(\\)]+[^\\.]$",
+      "x-ms-parameter-location": "method"
+    },
+    "apiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "description": "API Version",
+      "required": true,
+      "type": "string"
+    }
+  },
```

Comment:
There are definitions of common parameters in commontypes types.json that you can use by $ref instead of declaring your own, like SubscriptionIdParameter, OperationIdParameter, and ApiVersionParameter,
See
[https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L630](https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L630)

Reply:
Or if you had them in

 "$ref": "../../../Microsoft.Web/stable/2022-09-01/CommonDefinitions.json

you could use those I guess